### ISSUE TITLE
API for application support

### ISSUE THREAD

**Issue by CookiePLMonster** [nefarius/FireShock#5]
Will this feature a custom API to support all DS3/DS4 capabilities from applications? Incendium looks like it may be that in the future, but for now it's probably too unfinished to guess.

Ideally, the driver could allow running in both XInput (for most games) and custom mode so it works with games fine (like SCP does) but also allows for DS-specific support (like SCP's xinput extensions).

**Comment by mirh** [comment]
I guess this follows nefarius/ScpToolkit#82

**Comment by nefarius** [comment]
Yes there will be an API once I find the time to properly implement it.
